_posts/2022-08-11-putting-the-fun-in-funneljoin/putting-the-fun-in-funneljoin.Rmd

```diff
@@ -0,0 +1,289 @@
+
+
+## My first failure
+
+I dove into `after_join` with, basically, the default parameters. So let's break this down:
+
+* `x` is our procedures data because that's our reference point: we want ADT events *after* procedures
+* `y` is our ADT data to get those events after procedures
+* `by_user` is the column we use for identifiers (`ENCOUNTER_NUM_ANONYMIZED`); this is a very typical parameter to expect in `join` and `merge`, as we usually want to join data sets based on some identifier
+* `by_time` is the column we use to find the events (`EVENT_TS_FUZZED`); this is how `funneljoin` makes use of time series data
+* `suffix` appends labels to the newly created columns after the join, respectively for `x` then `y`
+
+Seems straight forward enough! Let's see what happens:
+
+```{r my_first_failure, echo = TRUE}
+
+library(funneljoin)
+
+join_attempt_one <- after_join(
+  x = SPECIFIC_PROCEDURES,
+  y = ADT_EVENTS,
+  by_user = "ENCOUNTER_NUM_ANONYMIZED",
+  by_time = "EVENT_TS_FUZZED",
+  suffix = c("_PROCEDURES","_ADT")
+)
+
+rmarkdown::paged_table(join_attempt_one)
+
+
+```
+
+
+Err... it sort of looks like nothing happens. Why is that? Well it's because our resulting join is empty^[again: I ain't so good at this]
+
+## My second failure
+
+I clearly missed something^[this is a generous way of saying I didn't read the documentation]. On a closer look at the parameters, it seemed like I needed a couple of more to make things work. So I brought them in and tried:
+
+* `mode` is `inner` which is a specific type of join which is basically *the intersection*
+* `type` here is one of many options on how to think about the join and we set it to `first-firstafter`
+
+Before we dive into the updated code let's pause for what `type` is and what `first-firstafter` means. Heavily borrowing from the `funneljoin` site: 
+
+*first-firstafter*: Take the first x, then the first y *after* that. For example, we have the first procedure for the first patient in the procedures (x) data, and we want the first event from the ADT (y) data that occurs afterwards. We don't want all afterward, we don't want any before. Just the one! So let's try it:
+
+```{r my_second_failure, echo = TRUE}
+
+join_attempt_two <- after_join(
+  x = SPECIFIC_PROCEDURES,
+  y = ADT_EVENTS,
+  by_user = "ENCOUNTER_NUM_ANONYMIZED",
+  by_time = "EVENT_TS_FUZZED",
+  suffix = c("_PROCEDURES","_ADT"),
+  mode = "inner",
+  type = "first-firstafter"
+)
+
+rmarkdown::paged_table(join_attempt_two)
+
+
+```
+
+The above shows the whole table but let's take a closer look at something...
```

Comment:
This shows up as empty for me 🤔 
![image](https://user-images.githubusercontent.com/4314491/184384215-8f2bb06b-e644-420e-9ea9-2e48f335c565.png)

Reply:
which version of R are you using? If it's 4.0.2 I think it'll be OK. If it's 3.6.3, then I believe it requires `stringsAsFactors = FALSE`. Which I should probably make explicit anyways...

Reply:
It works for me now!